src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,63 @@
+@import "./utils/_variables.scss";
+@import "./utils/_mixin.scss";
+
+.calendar {
+
+  display: flex;
+  justify-content: center;
+
+  &__box {
+    display: flex;
+    flex-wrap: wrap;
+    gap: $gap-size;
+
+    width: $calendar-width;
+    padding-top: 40px;
+    padding-left: center;
+
+    @for $i from 1 through 31 {
+      &--month-length--#{$i} {
+        & :nth-child(1n + #{$i+1}) {
+          display: none;
+        }
+      }
+    }
+
+    @each $day, $number in $days {
+      &--start-day--#{$day} {
+        & :first-child {
+          margin-left: ($number - 1) * ($day-size + 1px);
+        }
+      }
+    }
+  }
+
+  &__day {
+    box-sizing: border-box;
+
+    display: flex;
+    justify-content: center;
+    align-items: center;
+    text-align: center;
+
+    @include square($day-size);
+    border: 1px solid black;
+    background-color: $cell-bg;
+
+    transition: transform $transition-duration;
+
+    @for $i from 1 through 31 {
+      &--#{$i} {
+        &::after {
+          content: "#{$i}";
+
+        }
+      }
+    }
+    &:hover {
+      cursor: pointer;
```

Comment:
usually I indicate this on element themself, but it can be here, as for me it's bad practice, better to indicate it on class
also it's can't be animated